odbc/src/api.rs

```diff
@@ -0,0 +1,1047 @@
+#![allow(unused)]
+/// ^ TODO: remove that when implementations are complete
+use odbc_sys::{
+    BulkOperation, CDataType, Char, CompletionType, ConnectionAttribute, Desc, DriverConnectOption,
+    EnvironmentAttribute, FetchOrientation, HDbc, HDesc, HEnv, HStmt, HWnd, Handle, HandleType,
+    InfoType, Integer, Len, Nullability, ParamType, Pointer, RetCode, SmallInt, SqlDataType,
+    SqlReturn, StatementAttribute, ULen, USmallInt, WChar,
+};
+
+#[no_mangle]
+pub extern "C" fn SQLAllocHandle(
+    handle_type: HandleType,
+    input_handle: Handle,
+    output_handle: *mut Handle,
+) -> SqlReturn {
+    SqlReturn::ERROR
```

Comment:
```suggestion
    unimplemented!()
```
let's use `unimplemented!` for things that we haven't implemented yet, to differentiate from cases where we actually want to just return an error (like things we're going to intentionally not implement), with the idea being that all `unimplemented!` macro invocations have been replaced with something else by the time we close the epic